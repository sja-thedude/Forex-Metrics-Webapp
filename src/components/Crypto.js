import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { categories } from './data';
import ShowButton from './ShowButton';
import '../styles/Crypto.css';

const Crypto = () => {
  const cryptos = useSelector((state) => state.cryptos);
  const filterCryptos = cryptos.filter((crypto) => crypto !== undefined);
  const [isExpanded, setIsExpanded] = useState(false);
  const [itemsToShow, setItemsToShow] = useState(4);

  const navigate = useNavigate();

  const show = () => {
    if (itemsToShow === 4) {
      setItemsToShow(filterCryptos.length);
      setIsExpanded(true);
    } else {
      setItemsToShow(4);
      setIsExpanded(false);
    }
  };

  return (
    <section className="page-crypt">
      <div className="top-bar">
        <button
          type="button"
          className="go-back"
          onClick={() => navigate('/')}
        >
          <i className="chevron left icon" />
        </button>
        <h4>Crypto Currencies</h4>
        <div className="top-left" />
      </div>

      <div className="banner crypt">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'Cryptos' && (
          <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
            <div className="banner-info">
              <h3 className="banner-title">{category}</h3>
              <p className="total">{`Market Capitalization: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Tradable Crypto Pairs</h4>

      <ul className="crypt-pairs">
        { filterCryptos.slice(0, itemsToShow).map((pair) => {
          const {
            name, symbol, price, change,
          } = pair;

          return (
            <li key={symbol} className="crypt pair">
              <div className="crypt-info pair-name">
                <h3 className="crypt-symbol">{symbol}</h3>
                <p className="crypt-name">{name}</p>
              </div>

              <div className="pair-prices">
                <p>
                  <span className="pair-high">Price:</span>
                  {' '}
                  {price}
                </p>
                {
                  change > 0
                    ? (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="positive">
                          {`+${change.toFixed(5)} `}
                          &#9650;
                        </span>
                      </p>
                    )
                    : (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="negative">
                          {`${change.toFixed(5)} `}
                          &#9660;
                        </span>
                      </p>
                    )
                }
              </div>
            </li>
          );
        })}
      </ul>

      <ShowButton isExpanded={isExpanded} show={show} />
    </section>
  );
};

export default Crypto;
