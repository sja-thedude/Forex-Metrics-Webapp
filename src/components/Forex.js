import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import ShowButton from './ShowButton';
import { categories } from './data';
import '../styles/Forex.css';

const Forex = () => {
  const forex = useSelector((state) => state.forex);
  const [isExpanded, setIsExpanded] = useState(false);
  const [itemsToShow, setItemsToShow] = useState(3);
  const navigate = useNavigate();

  const show = () => {
    if (itemsToShow === 3) {
      setItemsToShow(forex.length);
      setIsExpanded(true);
    } else {
      setItemsToShow(3);
      setIsExpanded(false);
    }
  };

  return (
    <section className="page-forex">
      <div className="top-bar">
        <button
          type="button"
          className="go-back"
          onClick={() => navigate('/')}
        >
          <i className="chevron left icon" />
        </button>
        <h4>Currencies</h4>
        <div className="top-left" />
      </div>

      <div className="banner forex">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'Forex' && (
          <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
            <div className="banner-info">
              <h3 className="banner-title">{category}</h3>
              <p className="total">{`Market Capitalization: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Tradable Forex Pairs</h4>

      <ul className="forex-pairs">
        { forex.slice(0, itemsToShow).map((pair) => {
          const {
            ticker, high, low, changes,
          } = pair;

          return (
            <li key={ticker} className="forex pair">
              <h3 className="pair-name">{ticker}</h3>
              <div className="pair-prices">
                <p>
                  <span className="pair-high">High:</span>
                  {' '}
                  {high}
                </p>
                <p>
                  <span className="pair-low">Low:</span>
                  {' '}
                  {low}
                </p>
                {
                  changes > 0
                    ? (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="positive">
                          {`+${changes.toFixed(5)} `}
                          &#9650;
                        </span>
                      </p>
                    )
                    : (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="negative">
                          {`${changes.toFixed(5)} `}
                          &#9660;
                        </span>
                      </p>
                    )
                }
              </div>
            </li>
          );
        })}
      </ul>

      <ShowButton isExpanded={isExpanded} show={show} />
    </section>
  );
};

export default Forex;
