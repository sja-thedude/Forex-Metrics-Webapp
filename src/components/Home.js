import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { categories } from './data';
import {
  getForex, getCryptos, getStocks, getEFTs,
} from '../redux/actions/markets';
import '../styles/Home.css';

const Home = () => {
  const forex = useSelector((state) => state.forex);
  const cryptos = useSelector((state) => state.cryptos);
  const stocks = useSelector((state) => state.stocks);
  const efts = useSelector((state) => state.efts);

  const dispatch = useDispatch();

  useEffect(() => {
    if (!forex.length) {
      dispatch(getForex());
    }

    if (!cryptos.length) {
      setTimeout(() => dispatch(getCryptos()), 2000);
    }

    if (!stocks.length) {
      dispatch(getStocks());
    }

    if (!efts.length) {
      dispatch(getEFTs());
    }
  }, []);

  return (
    <section className="home-page">
      <div className="top-bar">
        <h4>Markets</h4>
        <div className="top-left">
          <i className="microphone icon" />
          <i className="cog icon" />
        </div>
      </div>

      <div className="banner home">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'Forex' && (
          <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
            <div className="banner-info">
              <h3 className="banner-title">{category}</h3>
              <p className="total">{`Market Capitalization: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Popular</h4>

      <div className="content-body">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category !== 'Forex' && (
          <Link key={path} to={`/${path}`} className={`category ${path}`}>
            <div className="category-info">
              <i className="chevron circle right icon" />
              <h3 className="category-title">{category}</h3>
              <p className="total">{`Cap: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <div className="bottom-bar" />
    </section>
  );
};

export default Home;
