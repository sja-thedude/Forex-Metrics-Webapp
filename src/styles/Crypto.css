.crypt-pairs {
  list-style: none;
  margin: 0;
  padding: 0;
}

.crypt.pair {
  height: 13.75vh;
}
