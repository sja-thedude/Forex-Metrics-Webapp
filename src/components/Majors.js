import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { categories } from './data';
import ShowButton from './ShowButton';
import { getMajors } from '../redux/actions/markets';
import '../styles/Forex.css';

const Majors = () => {
  const forex = useSelector((state) => state.forex);
  const majors = useSelector((state) => state.majors);
  const [isExpanded, setIsExpanded] = useState(false);
  const [itemsToShow, setItemsToShow] = useState(3);

  const navigate = useNavigate();
  const dispatch = useDispatch();

  useEffect(() => {
    if (forex.length && !majors.length) {
      dispatch(getMajors(forex));
    }
  }, []);

  const show = () => {
    if (itemsToShow === 3) {
      setItemsToShow(majors.length);
      setIsExpanded(true);
    } else {
      setItemsToShow(3);
      setIsExpanded(false);
    }
  };

  return (
    <section className="page-majors">
      <div className="top-bar">
        <button
          type="button"
          className="go-back"
          onClick={() => navigate('/')}
        >
          <i className="chevron left icon" />
        </button>
        <h4>Major Currencies</h4>
        <div className="top-left" />
      </div>

      <div className="banner majors">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'Majors' && (
          <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
            <div className="banner-info">
              <h3 className="banner-title">{category}</h3>
              <p className="total">{`Market Capitalization: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Tradable Forex Pairs</h4>

      <ul className="major-pairs">
        { majors.slice(0, itemsToShow).map((pair) => {
          const {
            ticker, high, low, changes,
          } = pair;

          return (
            <li key={ticker} className="majors pair">
              <h3 className="pair-name">{ticker}</h3>
              <div className="pair-prices">
                <p>
                  <span className="pair-high">High:</span>
                  {' '}
                  {high}
                </p>
                <p>
                  <span className="pair-low">Low:</span>
                  {' '}
                  {low}
                </p>
                {
                  changes > 0
                    ? (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="positive">
                          {`+${changes.toFixed(5)} `}
                          &#9650;
                        </span>
                      </p>
                    )
                    : (
                      <p>
                        <span className="pair-change">Change:</span>
                        {' '}
                        <span className="negative">
                          {`${changes.toFixed(5)} `}
                          &#9660;
                        </span>
                      </p>
                    )
                }
              </div>
            </li>
          );
        })}
      </ul>

      <ShowButton isExpanded={isExpanded} show={show} />
    </section>
  );
};

export default Majors;
