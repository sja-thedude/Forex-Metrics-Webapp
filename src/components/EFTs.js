import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import ShowButton from './ShowButton';
import { categories } from './data';
import Loading from './Loading';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import '../styles/EFTs.css';

const EFTs = () => {
  const efts = useSelector((state) => state.efts);
  const [isLoading, setIsLoading] = useState(true);
  const [isExpanded, setIsExpanded] = useState(false);
  const [itemsToShow, setItemsToShow] = useState(4);
  const navigate = useNavigate();

  useEffect(() => {
    if (itemsToShow === 4 || itemsToShow === efts.length) {
      setIsLoading(false);
    }
  }, [isExpanded]);

  const show = () => {
    if (itemsToShow === 4) {
      setItemsToShow(efts.length);
      setIsExpanded(true);
      setIsLoading(true);
    } else {
      setItemsToShow(4);
      setIsExpanded(false);
    }
  };

  return (
    <section className="page-efts">
      <div className="top-bar">
        <button
          type="button"
          className="go-back"
          onClick={() => navigate('/')}
        >
          <i className="chevron left icon" />
        </button>
        <h4>ETFs</h4>
        <div className="top-left" />
      </div>

      <div className="banner efts">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'ETF iShares' && (
            <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
              <div className="banner-info">
                <h3 className="banner-title">{category}</h3>
                <p className="total">{`Market Capitalization: ${cap}`}</p>
              </div>
            </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Tradable ETF iShares</h4>

      {
        isLoading
          ? <Loading body="eft-pairs" />
          : (
            <ul className="eft-pairs">
              { efts.slice(0, itemsToShow).map((eft) => {
                const { name, price, symbol } = eft;

                return (
                  <li key={symbol} className="efts pair">
                    <div className="eft-info pair-name">
                      <h3 className="eft-symbol">{symbol}</h3>
                      <p className="eft-name">{name}</p>
                    </div>

                    <div className="eft pair-prices">
                      <p>
                        <span className="pair-high">Price:</span>
                        {' '}
                        {price}
                      </p>
                    </div>
                  </li>
                );
              })}
            </ul>
          )
      }

      <ShowButton isExpanded={isExpanded} show={show} />
    </section>
  );
};

export default EFTs;
