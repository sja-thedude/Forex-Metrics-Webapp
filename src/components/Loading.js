import React from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import '../styles/EFTs.css';

function Loading({ body }) {
  return (
    <ul className={`${body} loading`}>
      <li>
        <Loader type="ThreeDots" color="var(--text-color)" height={50} width={50} />
      </li>
    </ul>
  );
}

Loading.propTypes = {
  body: PropTypes.string.isRequired,
};

export default Loading;
