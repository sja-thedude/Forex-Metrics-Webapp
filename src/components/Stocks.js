import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import ShowButton from './ShowButton';
import { categories } from './data';
import Loading from './Loading';
import '../styles/Stocks.css';

const Stocks = () => {
  const stocks = useSelector((state) => state.stocks);
  const [isLoading, setIsLoading] = useState(true);
  const [isExpanded, setIsExpanded] = useState(false);
  const [itemsToShow, setItemsToShow] = useState(4);
  const navigate = useNavigate();

  useEffect(() => {
    if (itemsToShow === 4 || itemsToShow === stocks.length) {
      setIsLoading(false);
    }
  }, [isExpanded]);

  const show = () => {
    if (itemsToShow === 4) {
      setItemsToShow(stocks.length);
      setIsExpanded(true);
      setIsLoading(true);
    } else {
      setItemsToShow(4);
      setIsExpanded(false);
    }
  };

  return (
    <section className="page-stocks">
      <div className="top-bar">
        <button
          type="button"
          className="go-back"
          onClick={() => navigate('/')}
        >
          <i className="chevron left icon" />
        </button>
        <h4>Stocks</h4>
        <div className="top-left" />
      </div>

      <div className="banner stocks">
        {categories.map((item) => {
          const { path, category, market_cap: cap } = item;

          return category === 'Stocks' && (
          <Link key={path} to={`/${path}`} className={`banner-link ${path}`}>
            <div className="banner-info">
              <h3 className="banner-title">{category}</h3>
              <p className="total">{`Market Capitalization: ${cap}`}</p>
            </div>
          </Link>
          );
        })}
      </div>

      <h4 className="sub-heading">Tradable Stocks</h4>

      {
        isLoading
          ? <Loading body="stock-pairs" />
          : (
            <ul className="stock-pairs">
              { stocks.slice(0, itemsToShow).map((stock) => {
                const { name, price, symbol } = stock;

                return (
                  <li key={symbol} className="stocks pair">
                    <div className="stock-info pair-name">
                      <h3 className="stock-symbol">{symbol}</h3>
                      <p className="stock-name">{name}</p>
                    </div>

                    <div className="stock pair-prices">
                      <p>
                        <span className="pair-high">Price:</span>
                        {' '}
                        {price}
                      </p>
                    </div>
                  </li>
                );
              })}
            </ul>
          )
      }

      <ShowButton isExpanded={isExpanded} show={show} />
    </section>
  );
};

export default Stocks;
