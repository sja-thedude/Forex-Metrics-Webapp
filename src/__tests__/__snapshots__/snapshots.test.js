import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Crypto from '../../components/Crypto';
import EFTs from '../../components/EFTs';
import Home from '../../components/Home';
import Forex from '../../components/Forex';
import Loading from '../../components/Loading';
import Majors from '../../components/Majors';
import Stocks from '../../components/Stocks';

describe('test screenshots for all components', () => {
  test('matches Crypto snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Crypto />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches EFTs snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <EFTs />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Forex snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Forex />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Loading snapshot', () => {
    const tree = TestRenderer.create(<Loading body="stocks" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Majors snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Majors />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Stocks snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Stocks />
        </Router>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
