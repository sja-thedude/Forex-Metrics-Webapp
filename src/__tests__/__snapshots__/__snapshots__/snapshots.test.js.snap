// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Crypto snapshot 1`] = `
<section
  className="page-crypt"
>
  <div
    className="top-bar"
  >
    <button
      className="go-back"
      onClick={[Function]}
      type="button"
    >
      <i
        className="chevron left icon"
      />
    </button>
    <h4>
      Crypto Currencies
    </h4>
    <div
      className="top-left"
    />
  </div>
  <div
    className="banner crypt"
  >
    <a
      className="banner-link crypto"
      href="/crypto"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          Cryptos
        </h3>
        <p
          className="total"
        >
          Market Capitalization: $2,78 T
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Tradable Crypto Pairs
  </h4>
  <ul
    className="crypt-pairs"
  />
  <button
    className="bottom-bar expand-button"
    onClick={[Function]}
    type="button"
  >
    More
     
    <i
      className="chevron down icon"
    />
  </button>
</section>
`;

exports[`test screenshots for all components matches EFTs snapshot 1`] = `
<section
  className="page-efts"
>
  <div
    className="top-bar"
  >
    <button
      className="go-back"
      onClick={[Function]}
      type="button"
    >
      <i
        className="chevron left icon"
      />
    </button>
    <h4>
      ETFs
    </h4>
    <div
      className="top-left"
    />
  </div>
  <div
    className="banner efts"
  >
    <a
      className="banner-link efts"
      href="/efts"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          ETF iShares
        </h3>
        <p
          className="total"
        >
          Market Capitalization: $325.43 B
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Tradable ETF iShares
  </h4>
  <ul
    className="eft-pairs loading"
  >
    <li>
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          fill="var(--text-color)"
          height={50}
          viewBox="0 0 120 30"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="15"
            cy="15"
            r={15}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="15"
              repeatCount="indefinite"
              to="15"
              values="15;9;15"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="1"
              repeatCount="indefinite"
              to="1"
              values="1;.5;1"
            />
          </circle>
          <circle
            attributeName="fillOpacity"
            cx="60"
            cy="15"
            from="1"
            r={9}
            to="0.3"
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="9"
              repeatCount="indefinite"
              to="9"
              values="9;15;9"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="0.5"
              repeatCount="indefinite"
              to="0.5"
              values=".5;1;.5"
            />
          </circle>
          <circle
            cx="105"
            cy="15"
            r={15}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="15"
              repeatCount="indefinite"
              to="15"
              values="15;9;15"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="1"
              repeatCount="indefinite"
              to="1"
              values="1;.5;1"
            />
          </circle>
        </svg>
      </div>
    </li>
  </ul>
  <button
    className="bottom-bar expand-button"
    onClick={[Function]}
    type="button"
  >
    More
     
    <i
      className="chevron down icon"
    />
  </button>
</section>
`;

exports[`test screenshots for all components matches Forex snapshot 1`] = `
<section
  className="page-forex"
>
  <div
    className="top-bar"
  >
    <button
      className="go-back"
      onClick={[Function]}
      type="button"
    >
      <i
        className="chevron left icon"
      />
    </button>
    <h4>
      Currencies
    </h4>
    <div
      className="top-left"
    />
  </div>
  <div
    className="banner forex"
  >
    <a
      className="banner-link forex"
      href="/forex"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          Forex
        </h3>
        <p
          className="total"
        >
          Market Capitalization: $6.6 T
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Tradable Forex Pairs
  </h4>
  <ul
    className="forex-pairs"
  />
  <button
    className="bottom-bar expand-button"
    onClick={[Function]}
    type="button"
  >
    More
     
    <i
      className="chevron down icon"
    />
  </button>
</section>
`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<section
  className="home-page"
>
  <div
    className="top-bar"
  >
    <h4>
      Markets
    </h4>
    <div
      className="top-left"
    >
      <i
        className="microphone icon"
      />
      <i
        className="cog icon"
      />
    </div>
  </div>
  <div
    className="banner home"
  >
    <a
      className="banner-link forex"
      href="/forex"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          Forex
        </h3>
        <p
          className="total"
        >
          Market Capitalization: $6.6 T
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Popular
  </h4>
  <div
    className="content-body"
  >
    <a
      className="category majors"
      href="/majors"
      onClick={[Function]}
    >
      <div
        className="category-info"
      >
        <i
          className="chevron circle right icon"
        />
        <h3
          className="category-title"
        >
          Majors
        </h3>
        <p
          className="total"
        >
          Cap: $6.6 T
        </p>
      </div>
    </a>
    <a
      className="category crypto"
      href="/crypto"
      onClick={[Function]}
    >
      <div
        className="category-info"
      >
        <i
          className="chevron circle right icon"
        />
        <h3
          className="category-title"
        >
          Cryptos
        </h3>
        <p
          className="total"
        >
          Cap: $2,78 T
        </p>
      </div>
    </a>
    <a
      className="category stocks"
      href="/stocks"
      onClick={[Function]}
    >
      <div
        className="category-info"
      >
        <i
          className="chevron circle right icon"
        />
        <h3
          className="category-title"
        >
          Stocks
        </h3>
        <p
          className="total"
        >
          Cap:  $48,56 M
        </p>
      </div>
    </a>
    <a
      className="category efts"
      href="/efts"
      onClick={[Function]}
    >
      <div
        className="category-info"
      >
        <i
          className="chevron circle right icon"
        />
        <h3
          className="category-title"
        >
          ETF iShares
        </h3>
        <p
          className="total"
        >
          Cap: $325.43 B
        </p>
      </div>
    </a>
  </div>
  <div
    className="bottom-bar"
  />
</section>
`;

exports[`test screenshots for all components matches Loading snapshot 1`] = `
<ul
  className="stocks loading"
>
  <li>
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        fill="var(--text-color)"
        height={50}
        viewBox="0 0 120 30"
        width={50}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="15"
          cy="15"
          r={15}
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="15"
            repeatCount="indefinite"
            to="15"
            values="15;9;15"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="1"
            repeatCount="indefinite"
            to="1"
            values="1;.5;1"
          />
        </circle>
        <circle
          attributeName="fillOpacity"
          cx="60"
          cy="15"
          from="1"
          r={9}
          to="0.3"
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="9"
            repeatCount="indefinite"
            to="9"
            values="9;15;9"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="0.5"
            repeatCount="indefinite"
            to="0.5"
            values=".5;1;.5"
          />
        </circle>
        <circle
          cx="105"
          cy="15"
          r={15}
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="15"
            repeatCount="indefinite"
            to="15"
            values="15;9;15"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="1"
            repeatCount="indefinite"
            to="1"
            values="1;.5;1"
          />
        </circle>
      </svg>
    </div>
  </li>
</ul>
`;

exports[`test screenshots for all components matches Majors snapshot 1`] = `
<section
  className="page-majors"
>
  <div
    className="top-bar"
  >
    <button
      className="go-back"
      onClick={[Function]}
      type="button"
    >
      <i
        className="chevron left icon"
      />
    </button>
    <h4>
      Major Currencies
    </h4>
    <div
      className="top-left"
    />
  </div>
  <div
    className="banner majors"
  >
    <a
      className="banner-link majors"
      href="/majors"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          Majors
        </h3>
        <p
          className="total"
        >
          Market Capitalization: $6.6 T
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Tradable Forex Pairs
  </h4>
  <ul
    className="major-pairs"
  />
  <button
    className="bottom-bar expand-button"
    onClick={[Function]}
    type="button"
  >
    More
     
    <i
      className="chevron down icon"
    />
  </button>
</section>
`;

exports[`test screenshots for all components matches Stocks snapshot 1`] = `
<section
  className="page-stocks"
>
  <div
    className="top-bar"
  >
    <button
      className="go-back"
      onClick={[Function]}
      type="button"
    >
      <i
        className="chevron left icon"
      />
    </button>
    <h4>
      Stocks
    </h4>
    <div
      className="top-left"
    />
  </div>
  <div
    className="banner stocks"
  >
    <a
      className="banner-link stocks"
      href="/stocks"
      onClick={[Function]}
    >
      <div
        className="banner-info"
      >
        <h3
          className="banner-title"
        >
          Stocks
        </h3>
        <p
          className="total"
        >
          Market Capitalization:  $48,56 M
        </p>
      </div>
    </a>
  </div>
  <h4
    className="sub-heading"
  >
    Tradable Stocks
  </h4>
  <ul
    className="stock-pairs loading"
  >
    <li>
      <div
        aria-busy="true"
        className=""
        style={Object {}}
      >
        <svg
          aria-label="audio-loading"
          fill="var(--text-color)"
          height={50}
          viewBox="0 0 120 30"
          width={50}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="15"
            cy="15"
            r={15}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="15"
              repeatCount="indefinite"
              to="15"
              values="15;9;15"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="1"
              repeatCount="indefinite"
              to="1"
              values="1;.5;1"
            />
          </circle>
          <circle
            attributeName="fillOpacity"
            cx="60"
            cy="15"
            from="1"
            r={9}
            to="0.3"
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="9"
              repeatCount="indefinite"
              to="9"
              values="9;15;9"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="0.5"
              repeatCount="indefinite"
              to="0.5"
              values=".5;1;.5"
            />
          </circle>
          <circle
            cx="105"
            cy="15"
            r={15}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="15"
              repeatCount="indefinite"
              to="15"
              values="15;9;15"
            />
            <animate
              attributeName="fillOpacity"
              begin="0s"
              calcMode="linear"
              dur="0.8s"
              from="1"
              repeatCount="indefinite"
              to="1"
              values="1;.5;1"
            />
          </circle>
        </svg>
      </div>
    </li>
  </ul>
  <button
    className="bottom-bar expand-button"
    onClick={[Function]}
    type="button"
  >
    More
     
    <i
      className="chevron down icon"
    />
  </button>
</section>
`;
